### PR TITLE
NIAD-2147: fix failing tests and merge with current main code ready f…

### PR DESCRIPTION
**Pull request by hodge70**
…or review.

Fixed violations, failing tests and affected main branch logic now appears to funciton as required.  All tests passed from local gradle run.

### CODE REVIEW

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java [hunk 1]
```diff
@@ -100,8 +102,14 @@ public void checkAndMergeFileParts(InboundMessage inboundMessage, String convers
...
         var currentAttachmentLog = patientAttachmentLogService.findAttachmentLog(inboundMessageId, conversationId);
+        var canCheckAttachmentLength = false;
+
+        if (currentAttachmentLog != null) {
+            canCheckAttachmentLength = true;
+        }
 
-        if (currentAttachmentLog == null) {
+        // if a message has arrived early, it will not have a parent ID so we can cancel early.
+        if (currentAttachmentLog == null || currentAttachmentLog.getParentMid() == null) {
```
Comment (benhession) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java:112]: There's a conflict here with the logic on line 123. Changes were brought in with NIAD-2156.

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java [hunk 2]
```diff
@@ -100,8 +102,14 @@ public void checkAndMergeFileParts(InboundMessage inboundMessage, String convers
         Document ebXmlDocument = xPathService.parseDocumentFromXml(inboundMessage.getEbXML());
         var inboundMessageId = xPathService.getNodeValue(ebXmlDocument, MESSAGE_ID_PATH);
         var currentAttachmentLog = patientAttachmentLogService.findAttachmentLog(inboundMessageId, conversationId);
+        var canCheckAttachmentLength = false;
```
Comment (benhession) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java:105]: Doesn't the check on line 112 ensure canCheckAttachmentLength is never false?

File: gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandlerTest.java [hunk 3]
```diff
@@ -506,11 +561,37 @@ public void When_CheckByteCompilationCreatesFileAsExpected_Expect_RunWithNoError
                 .thenReturn(createPatientAttachmentList(false, true, DATA_AMOUNT));
 
 
-        copcMessageHandler.checkAndMergeFileParts(inboundMessage, CONVERSATION_ID);
+
```
Comment (benhession) [inline gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandlerTest.java:564]: Have you left this whitespace here by mistake?

**Review by benhession — CHANGES_REQUESTED** [review]

File: gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandlerTest.java [hunk 4]
```diff
@@ -179,7 +183,7 @@ public void When_MIDFragmentRecordHasAnAttachmentLog_Expect_AttachmentLogToBeUpd
         // ACT
         copcMessageHandler.handleMessage(message, CONVERSATION_ID);
 
-        // ASSERT
+        //? ASSERT
```
Comment (mrogowska) [inline gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandlerTest.java:186]: why `//?`

**Review by mrogowska — COMMENTED** [review]
Checkstyle is failing: `/home/gradle/service/gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java:171:37: '{' is not preceded with whitespace.`

**Pull request closed without merging**